### PR TITLE
update binding details to design/persistent-storage doc

### PR DESCRIPTION
**Pull request by jeffvance**
Scroll down to "Matching and binding" (there is no TOC).
An issue I see is that this level of detail may get stale quickly and there is no built-in mechanism to update this doc when the binding code changes. However, several people on my team, from time to time, want to know greater details about binding rules, eg. for test writing, updating/creating other docs, etc.

PTAL @jsafrane @pmorie @copejon

If it's decided this type of content (high level overview of an algorithm/implementation) is useful, but docs/design/ is not the right location, perhaps a new dir can be created for this type of content?

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32817)

### CODE REVIEW

File: docs/design/persistent-storage.md [hunk 1]
```diff
@@ -229,6 +229,48 @@ NAME                LABELS              STATUS              VOLUME
...
+
+1. Otherwise, if the PV's "volume.beta.kubernetes.io/storage-class" annotation (which is a placeholder
+for a volume class) does *not* match the claim's annotation (same placeholder) then the PV is skipped.
+If the annotations for the PV and PVC are empty they are treated as being equal.
+
+1. Otherwise, what remains is a list of PVs that may match the claim. Within this list of remaining PVs,
+the PV with the smallest capacity that is also equal to or greater than the claim's requested storage
+is the matching PV and will be bound to the claim. Done. In the case of two or more PVCs matching all
+of the above criteria, the first PV (remember the PV order is based on `accessModes`) is the winner.
+
```
Comment (jsafrane) [inline docs/design/persistent-storage.md]: Please add a note that if no PV matches and a storage class is specified (or a default is set), then a volume is dynamically provisioned.

**Review by jsafrane — CHANGES_REQUESTED** [review]

**Comment by jsafrane** [comment]
Note that this PR does not "add" a new design, it updates the current one with actual implementation details.

**Comment by jeffvance** [comment]
@jsafrane _Note that this PR does not "add" a new design, it updates the current one with actual implementation details._
Agree. Changed pr title.  Thanks

**Comment by jeffvance** [comment]
@saad @eparis I've re-filled out the cla application so hopefully that issue is cleared now. @jsafrane can you LTGM ? Would very much like to get this pr merged and off my plate. Thanks!

**Comment by jsafrane** [comment]
@jeffvance, I'm waiting for this comment to be addressed: https://github.com/kubernetes/kubernetes/pull/32817#discussion_r79119998

After that, I'll be happy to merge this PR.

**Comment by jeffvance** [comment]
@jsafrane Sorry I missed your above comment. I just added the note. PTAL and hopefully now it is ok to merge. Thanks!

**Comment by jeffvance** [comment]
From error log: `E1019 08:53:14.668] FAIL: pull-kubernetes-e2e-gke-gci`.
@k8s-bot gci gke e2e test this

**Comment by jeffvance** [comment]
@jsafrane @saad-ali  Same CGI GKE smoke e2e error as above: `E1019 09:44:37.206] FAIL: pull-kubernetes-e2e-gke-gci`. Is it ok to ignore this error? If so then can this pr be merged now?  Thanks.

**Comment by jeffvance** [comment]
Ping @jsafrane @erinboyd @childsb can one of you PTAL? I believe this addition to the doc is still current -- Jan?

**Review by jsafrane — APPROVED** [review]

**Pull request merged**